From bba5b5dde3a9b27fdabf29e3d0b435c027a46605 Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Tue, 15 May 2018 12:42:49 -0500
Subject: [PATCH 1/5] FogBugz #559720: remove dummy client from service layer

commit b3fb5aa25d4ad26cd5be86ccc4b703718ffc6903 from
https://github.com/altera-opensource/linux-socfpga.git

Dummy client is not real service layer provider. It was created and used
to validate the communications among service provider, service layer
driver and secure monitor SW while we initially developed service layer
driver.

Dummy client is no longer need, and it should be removed from service
layer, since we have real service providers now (FPGA manager, RSU).

Signed-off-by: Richard Gong <richard.gong@intel.com>
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/misc/intel-service.c         |    7 +------
 include/linux/intel-service-client.h |    3 +--
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/misc/intel-service.c b/drivers/misc/intel-service.c
index 3019936..7bcdc3e 100644
--- a/drivers/misc/intel-service.c
+++ b/drivers/misc/intel-service.c
@@ -49,7 +49,7 @@
 /* SVC_NUM_DATA_IN_FIFO - number of struct intel_svc_data in the FIFO */
 #define SVC_NUM_DATA_IN_FIFO			32
 /* SVC_NUM_CHANNEL - number of channel supported by service layer driver */
-#define SVC_NUM_CHANNEL				3
+#define SVC_NUM_CHANNEL				2
 /*
  * FPGA_CONFIG_DATA_CLAIM_TIMEOUT_MS - claim back the submitted buffer(s)
  * from the secure world for FPGA manager to reuse, or to free the buffer(s)
@@ -928,11 +928,6 @@ static int intel_svc_drv_probe(struct platform_device *pdev)
 	chans[1].name = "rsu";
 	spin_lock_init(&chans[1].lock);
 
-	chans[2].scl = NULL;
-	chans[2].ctrl = controller;
-	chans[2].name = "dummy";
-	spin_lock_init(&chans[2].lock);
-
 	wake_up_process(controller->task);
 
 	list_add_tail(&controller->node, &svc_ctrl);
diff --git a/include/linux/intel-service-client.h b/include/linux/intel-service-client.h
index 4a1c3e9..604c3d4 100644
--- a/include/linux/intel-service-client.h
+++ b/include/linux/intel-service-client.h
@@ -9,10 +9,9 @@
 /*
  * Service layer driver supports client names
  * @fpga: for FPGA configuration
- * @dummy: for integration/debug/trouble-shooting
+ * @rsu: for remote status update
  */
 #define SVC_CLIENT_FPGA		"fpga"
-#define SVC_CLIENT_DUMMY	"dummy"
 #define SVC_CLIENT_RSU		"rsu"
 
 /*
-- 
1.7.9.5

